content/knowledge-base/import-variables-from-secret-manager.md

```diff
@@ -0,0 +1,395 @@
+      echo "DELIMITER" >> $CM_ENV
+        
+      echo "APP_STORE_CONNECT_KEY_IDENTIFIER=$(aws secretsmanager get-secret-value --secret-id APP_STORE_CONNECT_KEY_IDENTIFIER | jq -r '.SecretString')" >> $CM_ENV
+
+      echo "APP_STORE_CONNECT_ISSUER_ID=$(aws secretsmanager get-secret-value --secret-id APP_STORE_CONNECT_ISSUER_ID | jq -r '.SecretString')" >> $CM_ENV
+```
+
+Reference the variables for iOS publishing as usual:
+
+```
+publishing:
+  app_store_connect:              
+  api_key: $APP_STORE_CONNECT_PRIVATE_KEY      
+  key_id: $APP_STORE_CONNECT_KEY_IDENTIFIER     
+  issuer_id: $APP_STORE_CONNECT_ISSUER_ID
+```
+
+### Configuring an Android build using AWS secrets
+The environment variable `GCLOUD_SERVICE_ACCOUNT_CREDENTIALS` must be provided with a valid Service Account JSON key, even if you will overwrite it with a different key later. Add this variable to a group called `service_account` and then import it into you workflow as follows:
+
+```
+environment:
+  groups:
+    ...
+    - service_account
+    ...
+```
+
+Add the following environment variable in your codemagic.yaml:
+
+```
+environment:
+  groups:
+    ...
+  vars:
+    GCLOUD_SERVICE_ACCOUNT_CREDENTIALS_HOLDER: $GCLOUD_SERVICE_ACCOUNT_CREDENTIALS
+```
+
+In your first script step, retrieve the Service Account JSON from AWS and add it to **CM_ENV** as follows:
+
+```
+- name: Set Service Account JSON from AWS
+  script: |
+    echo "GCLOUD_SERVICE_ACCOUNT_CREDENTIALS<<DELIMITER" >> $CM_ENV
+    echo "$(aws secretsmanager get-secret-value --secret-id GCLOUD_SERVICE_ACCOUNT_CREDENTIALS | jq -r '.SecretString')" >> $CM_ENV
+          echo "DELIMITER" >> $CM_ENV
+```
+
+Reference the variable for publishing to Google Play as follows:
+
+```
+google_play:
+  credentials: $GCLOUD_SERVICE_ACCOUNT_CREDENTIALS_HOLDER
+  track: $GOOGLE_PLAY_TRACK
+```
+
+
+
+## Doppler Secrets Manager
+
+
+The following documentation shows how to do this using AWS Secrets Manager.
+
+### Add Doppler token to the Environment variables in Codemagic
+
+You will need to configure the environment variables `DOPPLER_TOKEN` in the Codemagic UI.
+
+Environment variables can be added at application level, or if you are using a Team, they can be added in the **Global variables and secrets** section of you Team settings.
+ 
+
+Add these variables to a group called `doopler_credentials`.
+
+### Storing a secret in Doppler Secrets 
+
+1. Log into your Doppler account and go to your workspace then your project.
+2. Navigate to the **SECRETS** tab,
+3. Click on the **Add Secret** button.
+4. Paste in the secret you would like to store, this can be an RSA certificate or simple string value.
+5. In the **NAME** section enter a name for your secret field. For example, if you are storing your App Store Connect API key, you might want to name it using the same naming convention as Codemagic, e.g. `APP_STORE_CONNECT_PRIVATE_KEY`.
+6. Repeat this process for other values you would like to store.
```

Comment:
```suggestion
### Storing a secret in Doppler 
To store a secret in Doppler, do the following:

1. Log into your Doppler account and go to your workspace then your project.
2. Navigate to the **SECRETS** tab,
3. Click on the **Add Secret** button.
4. Paste in the secret you would like to store, this can be an RSA certificate or simple string value.
5. In the **NAME** section enter a name for your secret field. For example, if you are storing your App Store Connect API key, you might want to name it using the same naming convention as Codemagic, e.g. `APP_STORE_CONNECT_PRIVATE_KEY`.
6. Repeat this process for other values you would like to store.
```